pkg/reconciler/instances/ory/hydra/hydra_test.go

```diff
@@ -0,0 +1,187 @@
+	t.Parallel()
+	t.Run("Should trigger synchronization if hydra-maester is behind hydra", func(t *testing.T) {
+		// given
+		hydraStartTimePod1 := time.Date(2021, 10, 10, 10, 10, 10, 10, time.UTC)
+		hydraStartTimePod2 := time.Date(2021, 10, 10, 10, 10, 7, 10, time.UTC)
+		hydraStartTimePod3 := time.Date(1900, 10, 10, 10, 10, 7, 10, time.UTC)
+		hydraMasesterPodStartTime := time.Date(2021, 10, 10, 10, 10, 6, 10, time.UTC)
+		kubeclient := fakeClient()
+		addPod(kubeclient, "hydra1", "hydra", hydraStartTimePod1, t, v1.PodRunning)
+		addPod(kubeclient, "hydra2", "hydra", hydraStartTimePod2, t, v1.PodRunning)
+		addPod(kubeclient, "hydra3", "hydra", hydraStartTimePod3, t, v1.PodFailed)
+		createDeployment(kubeclient, "ory-hydra-maester", hydraMasesterPodStartTime, t)
+		addPod(kubeclient, "hydra-maester1", "hydra-maester", hydraMasesterPodStartTime, t, v1.PodRunning)
+		client, _ := kubeclient.Clientset()
+
+		// when
+		err := NewDefaultHydraSyncer().TriggerSynchronization(context.TODO(), client, logger, "kyma-system")
+
+		// then
+		require.NoError(t, err)
+		deployment, err2 := client.AppsV1().Deployments(testNamespase).Get(context.TODO(), "ory-hydra-maester", metav1.GetOptions{})
+		require.NoError(t, err2)
+		spec := deployment.Spec
+		restartedAt := spec.Template.Annotations["kubectl.kubernetes.io/restartedAt"]
+		require.Greater(t, restartedAt, hydraMasesterPodStartTime.String())
+	})
+
+	t.Run("Should not trigger synchronization if hydra is behind hydra-maester", func(t *testing.T) {
+		// given
+		hydraStartTimePod1 := time.Date(2021, 10, 10, 10, 10, 10, 10, time.UTC)
+		hydraStartTimePod2 := time.Date(2021, 10, 10, 10, 10, 7, 10, time.UTC)
+		hydraStartTimePod3 := time.Date(2500, 10, 10, 10, 10, 6, 10, time.UTC)
+		hydraMasesterPodStartTime := time.Date(2022, 10, 10, 10, 10, 6, 10, time.UTC)
+		kubeclient := fakeClient()
+		addPod(kubeclient, "hydra1", "hydra", hydraStartTimePod1, t, v1.PodRunning)
+		addPod(kubeclient, "hydra2", "hydra", hydraStartTimePod2, t, v1.PodRunning)
+		addPod(kubeclient, "hydra3", "hydra", hydraStartTimePod3, t, v1.PodPending)
+		createDeployment(kubeclient, "ory-hydra-maester", hydraMasesterPodStartTime, t)
+		addPod(kubeclient, "hydra-maester", "hydra-maester", hydraMasesterPodStartTime, t, v1.PodRunning)
+		client, _ := kubeclient.Clientset()
+
+		// when
+		deploymentOld, _ := client.AppsV1().Deployments("kyma-system").Get(context.TODO(), "ory-hydra-maester", metav1.GetOptions{})
+		logger.Debugf("Deploy found ", deploymentOld.Name)
+		err := NewDefaultHydraSyncer().TriggerSynchronization(context.TODO(), client, logger, testNamespase)
+
+		// then
+		require.NoError(t, err)
+		deployment, err2 := client.AppsV1().Deployments(testNamespase).Get(context.TODO(), "ory-hydra-maester", metav1.GetOptions{})
+		require.NoError(t, err2)
+		spec := deployment.Spec
+		restartedAt := spec.Template.Annotations["kubectl.kubernetes.io/restartedAt"]
+		// deployment should not have been restarted
```

Comment:
Is this comment needed here? If it is, how about adding message to the `require.Equal()`? If it fails then we will know why exactly